corehq/form_processor/models.py

```diff
@@ -320,7 +320,10 @@ def to_json(self, include_attachments=False):
 
     def _get_attachment_from_db(self, attachment_name):
         from corehq.form_processor.backends.sql.dbaccessors import FormAccessorSQL
-        return FormAccessorSQL.get_attachment_by_name(self.form_id, attachment_name)
+        try:
```

Comment:
you could do this workflow only if the `form_id` has changed (see https://stackoverflow.com/a/1793323/632517 for a good way to determine that)